xlators/storage/posix/src/posix-inode-fd-ops.c

```diff
@@ -3349,9 +3348,8 @@ posix_links_in_same_directory(char *dirpath, int count, inode_t *leaf_inode,
     }
 
     while (count > 0) {
-        errno = 0;
```

Comment:
Are you sure errno is 0 before the system call?

Reply:
it doesn't matter, we don't require it afterwards.  The only time readdir() will *change* errno is if the return value is also NULL, so we just need to check the return value.  After this we execute closedir(), which will set errno if it returns -1.